### PR TITLE
Add backend method to download all chat history as json or markdown

### PR DESCRIPTION
**Pull request by KevinXuxuxu**
All messages in chat history are sorted chronologically.
To test download, do the following in browser console (e.g. for markdown format):
```
window.chatHistory.download('markdown');
```

Download data look something like this:
```json
[
  {
    "id": "a59a61d8-c3d3-4364-b00c-08a40791c834",
    "sid": "e37016f4-0711-4e36-aaea-315e8a74826c",
    "provider": "openai",
    "role": "Human",
    "content": "asdfe",
    "ts": "2024-04-28T22:03:43.102Z"
  },
  {
    "id": "15e64d90-28cb-4416-be68-62c6f009280a",
    "sid": "782cd576-9820-484e-8d0a-bf13e173d3e2",
    "provider": "openai",
    "role": "Human",
    "content": "Hello",
    "ts": "2024-04-28T22:21:28.690Z"
  },
  {
    "id": "00f67376-81e9-4836-aeb6-22812c3429d3",
    "sid": "782cd576-9820-484e-8d0a-bf13e173d3e2",
    "provider": "openai",
    "role": "AI",
    "content": "Hello there! How's your day going so far? If there's anything specific you'd like to chat about or any question you have in mind, feel free to ask. I'm here to help!",
    "ts": "2024-04-28T22:21:30.876Z"
  },
...
]
```
Rendered markdown result look like this:

##### Human
Hello!

Hello there! How’s your day going so far?

##### Human
So far so good! How about you?

I'm just a software, so I don't have feelings, but I'm here and ready to assist you! What's on your mind today that we can explore together?

##### Human
I'm working on this genAI based software that focuses on data privacy. Can you think of some ideas?

That sounds like a fascinating project! In terms of data privacy for your genAI software, you might consider implementing several strategies to enhance security and protect personal information. Here are some ideas:

1. **Data Anonymization and Encryption**: Before processing any data with genAI, anonymize personal identifiers to ensure that the data cannot be traced back to any individual. Encryption should also be employed both at rest and in transit to protect data from unauthorized access.

2. **Federated Learning**: This technique allows you to train an AI model across multiple decentralized devices or servers without exchanging data samples. This means the data remains in its original location and only relevant insights or model updates are shared. It could be particularly useful if you’re dealing with sensitive information spread across different jurisdictions.

3. **Differential Privacy**: Integrating differential privacy techniques can help in adding random noise to the data or queries to ensure the privacy of individual data points. This technique minimizes the risk of identifying personal data when the AI analyzes large datasets.
...

### CODE REVIEW

File: src/components/backend/utils.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const downloadData = (filename, data) => {
```
Comment (davidspu) [inline src/components/backend/utils.js:1]: 😂 is this the right way to do this...... impressive but feels wrong haha

Comment (KevinXuxuxu) [reply src/components/backend/utils.js:1]: I got this from chatgpt... Could you suggest a better way?

Comment (KevinXuxuxu) [reply src/components/backend/utils.js:1]: Adding a TODO and closing this comment.

**Review by lbyiuou0329 — APPROVED** [review]
lgtm

**Pull request merged**